pydantic_core/core_schema.py

```diff
@@ -3283,52 +3283,59 @@ def definition_reference_schema(
     return dict_not_none(type='definition-ref', schema_ref=schema_ref, metadata=metadata, serialization=serialization)
 
 
-CoreSchema = Union[
-    AnySchema,
-    NoneSchema,
-    BoolSchema,
-    IntSchema,
-    FloatSchema,
-    StringSchema,
-    BytesSchema,
-    DateSchema,
-    TimeSchema,
-    DatetimeSchema,
-    TimedeltaSchema,
-    LiteralSchema,
-    IsInstanceSchema,
-    IsSubclassSchema,
-    CallableSchema,
-    ListSchema,
-    TuplePositionalSchema,
-    TupleVariableSchema,
-    SetSchema,
-    FrozenSetSchema,
-    GeneratorSchema,
-    DictSchema,
-    AfterValidatorFunctionSchema,
-    BeforeValidatorFunctionSchema,
-    WrapValidatorFunctionSchema,
-    PlainValidatorFunctionSchema,
-    WithDefaultSchema,
-    NullableSchema,
-    UnionSchema,
-    TaggedUnionSchema,
-    ChainSchema,
-    LaxOrStrictSchema,
-    TypedDictSchema,
-    ModelSchema,
-    DataclassArgsSchema,
-    DataclassSchema,
-    ArgumentsSchema,
-    CallSchema,
-    CustomErrorSchema,
-    JsonSchema,
-    UrlSchema,
-    MultiHostUrlSchema,
-    DefinitionsSchema,
-    DefinitionReferenceSchema,
-]
+MYPY: Literal[False] = False
+
+
+if not MYPY:
```

Comment:
MyPy looks for the name `MYPY` and ignores `elif False`. Pyright needs the `elif False` to not complain about `CoreSchema` getting re-defined.